tls/extensions/s2n_client_alpn.c

```diff
@@ -35,3 +35,62 @@ int s2n_extensions_client_alpn_send(struct s2n_connection *conn, struct s2n_stuf
 
     return 0;
 }
+
+int s2n_recv_client_alpn(struct s2n_connection *conn, struct s2n_stuffer *extension)
+{
+    uint16_t size_of_all;
+    struct s2n_stuffer client_protos = {0};
+    struct s2n_stuffer server_protos = {0};
+
+    struct s2n_blob *server_app_protocols;
+    GUARD(s2n_connection_get_protocol_preferences(conn, &server_app_protocols));
+
+    if (!server_app_protocols->size) {
+        /* No protocols configured, nothing to do */
+        return 0;
+    }
+
+    GUARD(s2n_stuffer_read_uint16(extension, &size_of_all));
+    if (size_of_all > s2n_stuffer_data_available(extension) || size_of_all < 3) {
```

Comment:
Define a constant value for `3` here.

Reply:
This is just moving the logic over. from one file to another. removal of magic numbers will come at a later point in optimizing